src/Components/Components/src/RouteView.cs

```diff
@@ -17,6 +18,7 @@ namespace Microsoft.AspNetCore.Components;
 public class RouteView : IComponent
 {
     private RenderHandle _renderHandle;
+    private static readonly ConcurrentDictionary<Type, Type?> _layoutAttributeCache = new();
```

Comment:
I think we'll also need to clear this cache on hot reload, e.g., search for `MetadataUpdateSupported` and `OnDeltaApplied` in this project.

Reply:
Good point

Reply:
Can we avoid static caches like this and instead use a singleton in DI? IS that cumbersome for this?

Reply:
In this specific case it wouldn't be particularly difficult to inject a DI service here, so we could do.

However there are quite a lot of similar static caches elsewhere in the system, some of which may be in places where we'd have to do a lot more plumbing to get a DI service to them. I'm not how much to prioritize doing it differently for this particular case right away, versus reviewing them all collectively later.

Reply:
I don't think injecting is the hard part, it's that we would need to go through each host and add it to the collection, which is the painful part.

I'd rather keep this as it is right now and consider a different approach in the future.

Reply:
Just for clarification, I posted my comment before yours appeared on my screen, so it was intended as a reply to David rather than a reply to your comment. The order the messages appear now makes it seem like I was disagreeing with you but that is not at all the case!

Reply:
> However there are quite a lot of similar static caches elsewhere in the system, some of which may be in places where we'd have to do a lot more plumbing to get a DI service to them. I'm not how much to prioritize doing it differently for this particular case right away, versus reviewing them all collectively later.

Can you show me these other static caches?

Reply:
https://github.com/search?q=repo%3Adotnet%2Faspnetcore%20HotReloadManager.Default.MetadataUpdateSupported&type=code